File: leetcode/two-pointers/two_sum_ii.py
from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left, right = 0, len(numbers) - 1

        while left < right:
            total = numbers[left] + numbers[right]
            if total == target:
                return [left + 1, right + 1]
            elif total < target:
                left += 1
            else:
                right -= 1

        return []  # guaranteed to find a solution

# Try it out — press Run
sol = Solution()
print(sol.twoSum([1,2,3,4], 3))    # [1, 2]
print(sol.twoSum([1,3,4,5,7], 9))  # [3, 4]

File: leetcode/two-pointers/two_sum_ii.ts
function twoSum(numbers: number[], target: number): number[] {
  let left = 0;
  let right = numbers.length - 1;

  while (left < right) {
    const total = numbers[left] + numbers[right];
    if (total === target) return [left + 1, right + 1];
    else if (total < target) left++;
    else right--;
  }

  return [];
}

// Try it out — press Run
console.log(twoSum([1,2,3,4], 3));    // [1, 2]
console.log(twoSum([1,3,4,5,7], 9));  // [3, 4]

export {};
